### PR TITLE
Drop distro aliases and dot-notation tests (HMS-3710)

### PR DESCRIPTION
**Pull request by ochosi**
This reverts commit 2fe5e01d4d421d4d41e31f217c9b952d2e644aff.

This functionality is covered by unit tests in osbuild/images: https://github.com/osbuild/images/blob/aaa53adc3c1570bb54e65b15b9aa090991e74afb/pkg/distrofactory/distrofactory_test.go#L9

### CODE REVIEW

File: tools/provision.sh [hunk 1]
```diff
@@ -15,17 +15,6 @@ AUTH_METHOD_NONE="none"
...
-COMPOSER_CONFIG="/etc/osbuild-composer/osbuild-composer.toml"
-
-# Path to a file with additional configuration for composer.
-# The content of this file will be appended to the default configuration.
-EXTRA_COMPOSER_CONFIG="${2:-}"
-
-if [[ -n "${EXTRA_COMPOSER_CONFIG}" && ! -f "${EXTRA_COMPOSER_CONFIG}" ]]; then
-    echo "ERROR: File '${EXTRA_COMPOSER_CONFIG}' with extra configuration for composer does not exist."
-    exit 1
-fi
```
Comment (thozza) [inline tools/provision.sh:27]: The Weldr API test case depens on the changes in `provision.sh`: https://github.com/osbuild/osbuild-composer/blob/073e304978acade8bfa059f00a005402aa037c99/test/cases/weldr-distro-dot-notation-and-aliases.sh#L34-L43

**Review by thozza — CHANGES_REQUESTED** [review]
As far as I checked, it is OK to remove the `api.sh` test case, since all the aliasing and dot-notation backward compatibility is covered by unit tests in the `images` repo. In addtion, the alias configuration in composer is covered by unit tests in composer and the Weldr API test covers it as well (among other things, which are not covered by any unit tests).

However, the commit can't be simply reverted as is, since the Weldr API dot-notation/alias test case depends on the changes to `tools/provision.sh`.

File: tools/provision.sh (hunk 1 above)
Comment (ochosi) [reply tools/provision.sh]: Thanks for the review and for noticing :heart_eyes: 
Dropped those changes.

**Review by thozza — APPROVED** [review]
LGTM

**Pull request merged**